### PR TITLE
Remove fixme about LLVM basic block naming

### PR DESCRIPTION
**Pull request by erikdesjardins**
~This may be a small perf win.~

Originally, this PR implemented the fixme, but it didn't have any measurable perf improvement.

r? @ghost

### CODE REVIEW

File: compiler/rustc_codegen_llvm/src/builder.rs [hunk 1]
```diff
@@ -155,8 +155,14 @@ impl<'a, 'll, 'tcx> BuilderMethods<'a, 'tcx> for Builder<'a, 'll, 'tcx> {
 
     fn append_block(cx: &'a CodegenCx<'ll, 'tcx>, llfn: &'ll Value, name: &str) -> &'ll BasicBlock {
         unsafe {
-            let name = SmallCStr::new(name);
-            llvm::LLVMAppendBasicBlockInContext(cx.llcx, llfn, name.as_ptr())
+            let c_str_name;
+            let name_ptr = if cx.tcx.sess.fewer_names() {
```
Comment (klensy) [inline compiler/rustc_codegen_llvm/src/builder.rs:159]: Isn't `fewer_names` off by default, so need to revert this condition to actually test perf?

Comment (erikdesjardins) [reply compiler/rustc_codegen_llvm/src/builder.rs:159]: No, it's always enabled under normal circumstances. It's only set to false if you pass `--emit=llvm-ir`, or similar things that dump IR or assembly.

**Comment by erikdesjardins** [comment]
No effect; let's try again with the Rust side string formatting also avoided.

That has some small effect locally, but I'm not sure if it'll be significant:

```
> -1,519,050  <rustc_codegen_ssa::mir::FunctionCx<rustc_codegen_llvm::builder::Builder>>::try_llbb:???
> -1,509,872  core::fmt::write:???
>   -841,568  alloc::fmt::format::format_inner:???
```

There simply might not be enough bbs for this to matter.

**Comment by erikdesjardins** [comment]
No measurable change, there doesn't even seem to be a trend below the noise floor. I'll just delete the fixme.

**Pull request merged**